allocator_test.go

```diff
@@ -1,23 +1,239 @@
 package allocator_test
 
 import (
-	"testing"
-
 	"github.com/irfansharif/allocator"
 	"github.com/stretchr/testify/require"
+	"math/rand"
+	"testing"
 )
 
-func TestAllocator(t *testing.T) {
-	var rs []allocator.Range
-	for i := 0; i < 3; i++ {
-		rs = append(rs, allocator.NewRange(allocator.RangeID(i), 3, nil, nil))
+func TestGIVENReplicationConstraintTHENAppropriateReplicasAllocated(t *testing.T) {
```

Comment:
Why the random capitalization? These test names are too long + unreadable.

Reply:
Wanted to make the purpose of the test readable by the name itself, i.e `Test + GIVEN + test scenario + THEN + what we expect to happen`.